docs/zkapps/tutorials/09-recursion.mdx

```diff
@@ -117,103 +108,93 @@ We can also use recursion to combine two proofs:
+## Using ZkProgram in Smart Contracts
 
-Our zkRollup will operate over a MerkleMap of “accounts”, which each store a number. Our update rule will be to increment the value stored at an account - though you could imagine using more substantial rules to implement a particular application.
+After you build a recursive ZKP, use a method on a smart contract to settle the proof to the Mina blockchain. 
 
-The zkApp will store the Merkle root of this Merkle map of accounts on chain. We will allow this to be updated when authorized by a recursive zero knowledge proof generated by our ZkProgram.
+Build a zkRollup to use ZkProgram from a smart contract.
 
-This both allows it to be flexible to how much compute is being demanded of it (for latency), and allows it to scale to arbitrary numbers of proofs (for throughput).
+This example code builds a zkRollup to operate over a MerkleMap of accounts that each store a number. The update rule increments the value stored at an account - though you could imagine using more substantial rules to implement a particular application.
 
-Let’s discuss the code for the ZkProgram part of a rollup. You can find the full example [here](https://github.com/es92/zkApp-examples/blob/main/09-recursion/src/rollup.ts).
+The zkApp will store the Merkle root of this MerkleMap of accounts on chain. Updates occur only when authorized by a recursive zero knowledge proof generated by the ZkProgram.
 
-On one machine, the below code doesn’t offer a high throughput, but by switching the mapreduce used for a mapreduce that runs over tens or hundreds of machines, very high throughputs can be achieved. In fact, any level of throughput can be achieved as long as you’re willing to incur `log(N)` latency when constructing the proof of N transactions. You can find an implementation that distributes compute over AWS instances [here](https://github.com/Trivo25/proof_aggregator).
+This zkRollup design is flexible to how much compute is being demanded of it (for latency) and allows it to scale to arbitrary numbers of proofs (for throughput).
 
-To start, let’s set up our ZkProgram:
+On a single machine, the example code does not offer a high throughput. However, you can achieve very high throughputs by switching the mapreduce here for a mapreduce that runs over tens or hundreds of machines. In fact, you can achieve any level of throughput as long as you're willing to incur `log(N)` latency when constructing the proof of N transactions. 
+
+The following code for the ZkProgram part of a rollup is provided in the [rollup.ts](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/09-recursion/src/rollup.ts) example file.
+
+A community-contributed implementation distributes compute over AWS instances in this [proof_aggregator](https://github.com/Trivo25/proof_aggregator) example.
```

Comment:
@Trivo25  oooh! Look at you! did you want to move this example code to the docs repo? or leave as is?

Reply:
I built this thing for something other than the docs (it wasnt meant to be a tutorial or anything), so I think it's probably better to leave it in its own repository! But we can change the wording, call it an example or maybe even say its deprecated

Reply:
Nice @Trivo25 . I have a slight preference to move all code referenced in the docs to the docs repo so we can maintain it easier.

Reply:
The code in the docs is separate and not the same as in the linked package, so I don't think we have to move it over just because it's mentioned. If we decide to mention other community packages we also wouldn't want to move their code into the docs repository. I am good with removing the link to my package if that is a better solution, I think the original author put it there as a reference when that section was written

Reply:
I'm ok with leaving as is @Trivo25